### PR TITLE
1443 cleanup fix

### PR DESCRIPTION
**Pull request by JohnPTobe**
Fixes for these issues: #1443, #1444, #1448

When a release is made with these changes, the release notes need to reflect the new required version of docker (1.13). See #1447

### CODE REVIEW

File: scale/scheduler/node/conditions.py [hunk 1]
```diff
@@ -123,20 +134,48 @@ def handle_cleanup_task_completed(self):
         """
 
         self._error_inactive(NodeConditions.CLEANUP_ERR)
+        self._warning_inactive_old()
         self._update_state()
 
-    def handle_cleanup_task_failed(self):
+    def handle_cleanup_task_failed(self, job_exes):
```
Comment (gisjedi) [inline scale/scheduler/node/conditions.py]: This is a good change. Can we get a unit test to validate the new method?

File: scale/scheduler/node/conditions.py [hunk 2]
```diff
@@ -123,20 +134,48 @@ def handle_cleanup_task_completed(self):
...
+            title = NodeConditions.CLEANUP_FAILURE.title
+            description = NodeConditions.CLEANUP_FAILURE.description % ids
+            self._warning_active(NodeWarning(name=name, title=title), description)
+        else:
+            logger.warning('Cleanup task failed with no job exes')
+        
         self._update_state()
 
-    def handle_cleanup_task_timeout(self):
+    def handle_cleanup_task_timeout(self, job_exes):
```
Comment (gisjedi) [inline scale/scheduler/node/conditions.py]: This is a good change. Can we get a unit test to validate the new method?

**Comment by JohnPTobe** [comment]
Done with changes, assuming tests pass

**Comment by JohnPTobe** [comment]
This has been tested in dev and cleanups seem to be working. I haven't been able to trigger a cleanup error to test the warnings.

**Comment by JohnPTobe** [comment]
Closes #1443, closes #1444, closes #1448

**Comment by JohnPTobe** [comment]
Closes #1444

**Comment by JohnPTobe** [comment]
Closes #1448

**Pull request merged**